grpc/src/compiler/python_generator.cc

```diff
@@ -136,14 +157,16 @@ class StubGenerator : public BaseGenerator {
   bool Generate() {
     Imports imports;
     std::stringstream stub;
+    std::string ns_name{};
     for (const ServiceDef *service : parser_.services_.vec) {
       Generate(stub, service, &imports);
+      ns_name = NamespaceDir(*service->defined_namespace, false);
     }
 
     std::string sanitized_suffix{ parser_.opts.grpc_filename_suffix };
     std::replace(sanitized_suffix.begin(), sanitized_suffix.end(), '.', '_');
     std::string filename =
-        namer_.config_.output_path +
+        namer_.config_.output_path + ns_name + kPathSeparator +
```

Comment:
Can this be done outside of the generator and passed in `namer_.config_.output_path`? This will not work with our Google-internal setup as we need to know or be able to override the path to the generated file without looking into the content of the original `.fbs` file.

Reply:
Assuming using --gen-onefile is an option this would work now ->

```
../flatc --gen-onefile --grpc --python -o foo ./portal.fbs
ls foo/portal_grpc_fb.py 
```

eg the path is -o `foo` and the filename is filename of the `.fbs` file + _grpc_fb.py


Otherwise it's a bit awkward to parse the file first to get the namespaces and then again later.
But if `--gen-onefile` is not possible, we need to look into that

Reply:
`--gen-onefile` is what we actually use. If that still works, it's fine with me!

Reply:
ok yeah as far as i understand your usecase this should be fine as is 👍